hmiddleware/httpmetrics/httpmetrics_v2.go

```diff
@@ -0,0 +1,74 @@
+package httpmetrics
+
+import (
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/heroku/x/go-kit/metrics"
+	"github.com/heroku/x/go-kit/metricsregistry"
+
+	"github.com/go-chi/chi"
+	"github.com/go-chi/chi/middleware"
+)
+
+const (
+	// metric names
+	requestDuration = "http.server.duration"        // duration in milliseconds
+	activeRequests  = "http.server.active_requests" // counter for number of requests
+
+	// metric attribute keys
+	routeKey         = "http.route"
+	methodKey        = "http.request.method"
+	statusKey        = "http.response.status_code"
+	serverAddressKey = "server.address"
+	urlSchemeKey     = "url.scheme"
+)
+
+// NewV2 returns an HTTP middleware which captures HTTP request counts and latency
+// annotated with attributes for method, route, status.
+//
+// See https://opentelemetry.io/docs/specs/otel/metrics/semantic_conventions/http-metrics/
+func NewV2(p metrics.Provider) func(http.Handler) http.Handler {
+	reg := metricsregistry.New(p)
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			ww := middleware.NewWrapResponseWriter(w, r.ProtoMajor)
+
+			start := time.Now()
+			next.ServeHTTP(ww, r)
+			dur := time.Since(start)
+
+			labels := []string{
+				methodKey, r.Method,
+			}
+
+			if status := ww.Status(); status != 0 {
+				kv := []string{statusKey, strconv.Itoa(status)}
+				labels = append(labels, kv...)
+			}
+
+			ctx := r.Context()
+			if ctx.Value(chi.RouteCtxKey) != nil {
+				rtCtx := chi.RouteContext(ctx)
+				if len(rtCtx.RoutePatterns) > 0 {
+					// pick last route pattern as it is the one chi used
+					route := rtCtx.RoutePatterns[len(rtCtx.RoutePatterns)-1]
+					kv := []string{routeKey, route}
+					labels = append(labels, kv...)
+				}
+			}
+
+			if r.URL != nil {
+				kv := []string{
+					urlSchemeKey, r.URL.Scheme,
+					serverAddressKey, r.URL.Host,
+				}
+				labels = append(labels, kv...)
+			}
+
+			reg.GetOrRegisterCounter(activeRequests).With(labels...).Add(1)
```

Comment:
This will change the meaning of [`http.server.active_requests`](https://opentelemetry.io/docs/specs/otel/metrics/semantic_conventions/http-metrics/#metric-httpserveractive_requests). I know this pattern is matching our current usage with the `http.server.requests` metric, but I suggest tweaking it to match the intended purpose. It should track the active in-flight requests. This will avoid confusion down the road when we standardize on metric naming conventions.

On the line below, we report the metric `http.server.duration` as a histogram. This will automatically make additional metrics available, including:
```
http.server.duration.count
http.server.duration.sum
http.server.duration.avg
http.server.duration.min
http.server.duration.max
http.server.duration.p001...p999
```

Since we will need to update dashboards anyway, I would suggest relying on `http.server.duration.count` instead of our older atomic counter style independent metric.

Reply:
Agree, I thought about dropping the `http.server.active_requests` as it is optional anyway. I wondered if `http.server.active_requests` is useful considering that we report on a timer, many values just get dropped which might skew the picture. 

I'm ok with using `http.server.duration.count` as the standard for counting requests.

Reply:
I'm going to drop `http.server.active_requests` for now. We can determine most attributes only when the request is complete. We can rethink this if we need it in the future.

Reply:
WFM

Reply:
Thank you